syntax = "proto3";

package hil_serl;

// LearnerService: the Actor calls this to push transitions.
// The Learner implements this service.
service LearnerService {
  // Actor -> Learner to store transitions
  rpc SendTransition(Transition) returns (Empty);
  rpc SendInteractionMessage(InteractionMessage) returns (Empty);  
}

// ActorService: the Learner calls this to push parameters.
// The Actor implements this service.
service ActorService {
  // Learner -> Actor to send new parameters
  rpc StreamTransition(Empty) returns (stream ActorInformation) {};
  rpc SendParameters(Parameters) returns (Empty);
}


message ActorInformation {
    oneof data {
        Transition transition = 1;
        InteractionMessage interaction_message = 2;
    }
}

// Messages
message Transition {
  bytes transition_bytes = 1;
}

message Parameters {
  bytes parameter_bytes = 1;
}

message InteractionMessage {
  bytes interaction_message_bytes = 1; 
}

message Empty {}